api/query_engine.go

```diff
@@ -61,47 +62,60 @@ var (
 //    For "fairness" across series, and because we used to simply reduce any series without regard for how it would be used, we pick the latter. better would be both
 // 3) subject to max-points-per-req-hard: reject the query if it can't be met
 //
-// note: it is assumed that all requests have the same from & to.
 // also takes a "now" value which we compare the TTL against
 
 // TODO: MDP-yes and max-points-per-req-soft code paths may not take into account that archive 0 may have a different raw interval.
 // see https://github.com/grafana/metrictank/issues/1679 (for MDP-no it does do the right thing)
-func planRequests(now, from, to uint32, reqs *ReqMap, planMDP uint32, mpprSoft, mpprHard int) (*ReqsPlan, error) {
+func planRequests(now uint32, reqs *ReqMap, planMDP uint32, mpprSoft, mpprHard int) (*ReqsPlan, error) {
 
 	ok, rp := false, NewReqsPlan(*reqs)
 
 	// 1) Initial parameters
+	getTimeWindowSuperSet := func(reqs ReqsByRet) (uint32, uint32) {
+		minFrom := uint32(math.MaxUint32)
+		maxTo := uint32(0)
+		for _, rs := range reqs {
+			for _, r := range rs {
+				minFrom = util.Min(minFrom, r.From)
+				maxTo = util.Max(maxTo, r.To)
+			}
+		}
+		return minFrom, maxTo
+	}
 	for group, split := range rp.pngroups {
+		// Find the minFrom and maxTo of reqs in the same split
```

Comment:
each PNGroup by definition corresponds to the same target. Because that's what pre-normalizing is for: assuring data that will go into the same sumSeries (for example) function, gets the same resolution.
This normally means for a given PNGroup, all requests have the same from/to - except maybe in a case like `sum(timeStack(..))`

Reply:
Ah, right. I can probably just take the first one in this case.

Reply:
yes but not for the `sum(timeStack(..))` case. I'm pondering maybe forbidding this type of query but it's not an unreasonable one, so we should try to accommodate it.

Reply:
I expect `timeStack` will be a significantly different beast altogether. I don't plan on implementing it any time soon :/

FWIW, I see `avg(timeStack(...))` pop up reasonably frequently to compare the current data against the avg of some number of intervals. It can be useful.

Reply:
> I expect timeStack will be a significantly different beast altogether. I don't plan on implementing it any time soon 

that's fine. we can just create a stub expr/func_timestack.go and put any notes in there we may want to leave for our future selves

Reply:
Looks like we still do the getTimeWindowSuperSet calls even for PNGroups. as mentioned before, this is unnessecary, except in the case of `sum(timeStack(..))` style queries. I suppose we can leave it as is, to make it easier on ourselves when we do implement timeStack.

Reply:
It doesn't cost us much to leave it and overall seems like one less piece of context to remember.